src/hooks/useDragAndDrop.ts

```diff
@@ -86,7 +89,7 @@ export default function useDragAndDrop({dropZone, onDrop = () => {}, shouldAllow
                         return;
                     }
                     // This is necessary because dragging over children will cause dragleave to execute on the parent.
-                    if ((event.currentTarget as HTMLElement | null)?.contains(event.relatedTarget as HTMLElement | null)) {
+                    if (enterTarget.current !== event.target){
```

Comment:
What if the enter target gets removed? I think we are relying on another element to take the enter target role?

Reply:
I can't find a particular bug here. Let's go with this